src/udev/udev-event.c

```diff
@@ -681,9 +685,14 @@ static int spawn_wait(Spawn *spawn) {
                         return r;
         }
 
-        r = sd_event_add_child(e, NULL, spawn->pid, WEXITED, on_spawn_sigchld, spawn);
+        r = sd_event_add_child(e, &s, spawn->pid, WEXITED, on_spawn_sigchld, spawn);
         if (r < 0)
                 return r;
+        /* SIGCHLD should be processed after IO is complete */
+        r = sd_event_source_set_priority(s, SD_EVENT_PRIORITY_NORMAL + 1);
+        if (r < 0)
+                return r;
```

Comment:
this leaks the event source object. If you acquire a reference to it, you need to either keep it around as long as you need it and then unref it or mark the even source as "floating", in which case it is released automatically when the event source is freed. i.e. easiest is probably to use `_cleanup_(sd_event_source_unrefp) …` on the variable declaration. Or alternatively use sd_event_source_set_floating()

Reply:
I will fix this tomorrow.